internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,699 @@
+            }
+        });
+    }
+
+    fn arm_value_tracker(self: Pin<&Self>) {
+        let item = self.item.clone();
+        let p = self.project_ref();
+        p.value_tracker.evaluate_as_dependency_root(move || {
+            if let Some(item_rc) = item.upgrade() {
+                item_rc.accessible_string_property(AccessibleStringProperty::Value);
+                item_rc.accessible_string_property(AccessibleStringProperty::ValueMinimum);
+                item_rc.accessible_string_property(AccessibleStringProperty::ValueMaximum);
+                item_rc.accessible_string_property(AccessibleStringProperty::ValueStep);
+            }
+        });
+    }
+
+    fn arm_label_tracker(self: Pin<&Self>) {
+        let item = self.item.clone();
+        let p = self.project_ref();
+        p.label_tracker.evaluate_as_dependency_root(move || {
+            if let Some(item_rc) = item.upgrade() {
+                item_rc.accessible_string_property(AccessibleStringProperty::Label);
+            }
+        });
+    }
+
+    fn arm_description_tracker(self: Pin<&Self>) {
+        let item = self.item.clone();
+        let p = self.project_ref();
+        p.description_tracker.evaluate_as_dependency_root(move || {
+            if let Some(item_rc) = item.upgrade() {
+                item_rc.accessible_string_property(AccessibleStringProperty::Description);
+            }
+        });
+    }
+
+    fn arm_focus_delegation_tracker(self: Pin<&Self>) {
+        let item = self.item.clone();
+        let p = self.project_ref();
+        p.focus_delegation_tracker.evaluate_as_dependency_root(move || {
+            if let Some(item_rc) = item.upgrade() {
+                item_rc.accessible_string_property(AccessibleStringProperty::DelegateFocus);
+            }
+        });
+    }
+}
+
+cpp! {{
+    #include <QtWidgets/QtWidgets>
+
+    #include <memory>
+
+    /// KEEP IN SYNC WITH CONSTANTS IN RUST!
+    const uint32_t NAME { QAccessible::Name };
+    const uint32_t DESCRIPTION { QAccessible::Description };
+    const uint32_t VALUE { QAccessible::Value};
+    const uint32_t CHECKED { QAccessible::UserText };
+    const uint32_t VALUE_MINIMUM { CHECKED + 1 };
+    const uint32_t VALUE_MAXIMUM { VALUE_MINIMUM + 1 };
+    const uint32_t VALUE_STEP { VALUE_MAXIMUM + 1 };
+
+    // ------------------------------------------------------------------------------
+    // Helper:
+    // ------------------------------------------------------------------------------
+
+    class Descendents {
+    public:
+        Descendents(void *root_item) {
+            rustDescendents = rust!(Descendents_ctor [root_item: *mut c_void as "void*"] ->
+                    SharedVector<ItemRc> as "void*" {
+                let mut descendents = i_slint_core::accessibility::accessible_descendents(
+                        &*(root_item as *mut ItemRc));
+                SharedVector::from_slice(&descendents)
```

Comment:
In combination with the change to `accessible_descendents()` I think that this may avoid the repeated allocation of new vectors and the shared vector at the end, in favour of just one single shared vector.

```suggestion
                let mut descendents = SharedVector::default();
                i_slint_core::accessibility::accessible_descendents(
                        &*(root_item as *mut ItemRc), &mut descendents);
                descendents
```